frontend/src/main.js

```diff
@@ -2,11 +2,43 @@ import Vue from 'vue'
 import App from './App'
 import router from './router'
 import store from './store'
+import injectInitialState from './utils/inject-state'
 import './config'
 
-new Vue({ // eslint-disable-line no-new
-  el: '#app',
+const app = new Vue({ // eslint-disable-line no-new
   router,
   store,
   render: h => h(App)
 })
+
+// reference: https://markus.oberlehner.net/blog/how-to-pre-render-vue-powered-websites-with-webpack/
+
+// During pre-rendering the initial state is
+// injected into the global scope, here we
+// fill the store with the initial state.
+if (window.__INITIAL_STATE__) store.replaceState(window.__INITIAL_STATE__)
+
+router.beforeResolve(async (to, from, next) => {
+  try {
+    const components = router.getMatchedComponents(to)
+
+    // By using `await` we make sure to wait
+    // for the API request made by the `fetch()`
+    // method to resolve before rendering the view.
+    await Promise.all(components.map(x => x.fetch && x.fetch({ store })))
```

Comment:
I don't get this - is it waiting to resolve fetch()es that exist in the created() hook for each component or something?

Reply:
tbh at the moment this does nothing.

It's waiting for the resolution of an async function called fetch inside each component.

More here: https://markus.oberlehner.net/blog/how-to-pre-render-vue-powered-websites-with-webpack/

Reply:
In theory, yes. In doing so it would await things like API calls to resolve before rendering. To go full SSR you can do these API calls server side and not have to do again client side.